InterruptServer.rb

```diff
@@ -0,0 +1,212 @@
+require 'socket'
+require 'json'
+
+class InterruptServer
+
+	CHAT_LENGTH = 45 # length of chat text user can see at one time
+	TICK_LENGTH = 2 # how many seconds to wait before 'moving chat text left'
+	MAX_MSG_LENGTH = 1024 # max length of incoming message read
+	
+	def initialize(host, port, num_colors=5)
+		@server = UDPSocket.new
+		@server.bind(host, port)
+
+		@clients = {}
+		@colors_available = [*0..num_colors-1]
```

Comment:
Does the server actually know about colors? It doesn't seem like something that really needs to be baked into the protocol or server. If this is just an integer, it would be more aptly named as a client "id" or "number" or something. The notion of color would only exist on the client side. One could imagine an alternative client that uses some other way of differentiating clients instead of color.